redis/src/sentinel.rs

```diff
@@ -295,25 +295,57 @@ fn valid_addrs<'a>(
         })
 }
 
-fn check_role_result(result: &RedisResult<Vec<Value>>, target_role: &str) -> bool {
-    if let Ok(values) = result {
-        if !values.is_empty() {
-            if let Ok(role) = String::from_redis_value(&values[0]) {
-                return role.to_ascii_lowercase() == target_role;
-            }
-        }
+fn determine_master_from_role_or_info_replication(
+    connection_info: &ConnectionInfo,
+) -> RedisResult<bool> {
+    let client = Client::open(connection_info.clone())?;
+    let mut conn = client.get_connection()?;
+
+    //Once the client discovered the address of the master instance, it should attempt a connection with the master, and call the ROLE command in order to verify the role of the instance is actually a master.
+    let role = check_role(&mut conn);
+    if role.is_ok_and(|x| matches!(x, Role::Primary { .. })) {
```

Comment:
the only error which is actually ok to continue here is unknown command (for redis 2), right? so maybe check if that's the error before continuing?

Reply:
Thats incorrect. The other error possible is NOPERM.

Reply:
In my env, they default to no role permissions but allow for the replication.